### PR TITLE
[HOTFIX][#3705] fix(spark-connector): write data to Iceberg table failed if drop and recreate Iceberg table 

### PR DESCRIPTION
**Pull request by FANNG1**
### What changes were proposed in this pull request?
disable Iceberg catalog cache

### Why are the changes needed?

Fix: #3705 

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
add IT

### CODE REVIEW

**Comment by FANNG1** [comment]
If enable catalog cache, drop table is just through gravitino and couldn't invalid table cache in internal spark catalog. If creating tables with same name, the table info from the cache is not correct. This is a quick fix by disabling catalog cache in spark connector, a more general way maybe though Gravitino in #3707

**Comment by FANNG1** [comment]
@jerryshao @qqqttt123 @caican00 please help to review when you are free, thanks

**Comment by jerryshao** [comment]
So this is just a hotfix, right?

File: spark-connector/spark-common/src/main/java/com/datastrato/gravitino/spark/connector/iceberg/SparkIcebergTable.java [hunk 1]
```diff
@@ -37,7 +36,7 @@ public SparkIcebergTable(
       PropertiesConverter propertiesConverter,
       SparkTransformConverter sparkTransformConverter,
       SparkTypeConverter sparkTypeConverter) {
-    super(sparkTable.table(), !isCacheEnabled(sparkCatalog));
+    super(sparkTable.table(), true);
```
Comment (jerryshao) [inline spark-connector/spark-common/src/main/java/com/datastrato/gravitino/spark/connector/iceberg/SparkIcebergTable.java:39]: Can you please write like  this`super(sparkTable.table(), true /** refreshEagerly */)` ?

Comment (FANNG1) [reply spark-connector/spark-common/src/main/java/com/datastrato/gravitino/spark/connector/iceberg/SparkIcebergTable.java]: done

**Comment by FANNG1** [comment]
> So this is just a hotfix, right?

yes, I think this configuration is better managed by Gravitino, and there are similar problems for Gravitino Iceberg catalog if enabling catalog cache and using multi Gravitino server.

**Review by jerryshao — APPROVED** [review]

**Pull request merged**